chain/db/stateGenerator.go

```diff
@@ -120,6 +121,25 @@ func (cs *ChainStore) generateStateRoot(b *block.Block, genesisBlockInitialized,
 
 	//process previous block
 	height := b.Header.UnsignedHeader.Height
+	if height == 0 {
+		preBlockHash := EmptyUint256
```

Comment:
What about creating a common function like ComputeID(preBlockHash, txnHash, randomBeacon) to share between here and below (around line 153)?

Reply:
good idea.